recipes/fastpfor/all/patches/0.2.0-0001-fix-cmake.patch

```diff
@@ -0,0 +1,99 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 38590f3..cbe7357 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -62,7 +62,7 @@ if( SUPPORT_SSE42 )
+     MESSAGE( STATUS "SSE 4.2 support detected" )
+ else()
+     if (SUPPORT_NEON)
+-        include("${CMAKE_MODULE_PATH}/simde.cmake")
++        find_package(simde REQUIRED CONFIG)
+         MESSAGE(STATUS "USING SIMDE FOR SIMD OPERATIONS")
+     else ()
+         MESSAGE(STATUS "SIMDE and SSE 4.2 support not detected")
+@@ -90,7 +90,7 @@ elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
+     set (CMAKE_CXX_FLAGS_DEBUG   "-Wall -ggdb   -std=c++11 -DHAVE_CXX0X  -march=native")
+     set (CMAKE_C_FLAGS_RELEASE "-Wall -Ofast  -DNDEBUG -std=c99  -march=native")
+     set (CMAKE_C_FLAGS_DEBUG   "-Wall -ggdb   -std=c99  -march=native")
+-elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang" OR ${CMAKE_CXX_COMPILER_ID} STREQUAL "AppleClang")
++elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
+     if (CXX_COMPILER_VERSION VERSION_LESS 4.2.1)
+         message(STATUS  "Clang version must be at least 4.2.1!" )
+     endif()
+@@ -98,11 +98,24 @@ elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang" OR ${CMAKE_CXX_COMPILER_ID} STR
+     set (CMAKE_CXX_FLAGS_DEBUG   "-Wall -Wcast-align -ggdb  -std=c++11 -DHAVE_CXX0X -march=native")
+     set (CMAKE_C_FLAGS_RELEASE "-Wall -Wcast-align -O3 -DNDEBUG -std=c99 -march=native")
+     set (CMAKE_C_FLAGS_DEBUG   "-Wall -Wcast-align -ggdb  -std=c99 -march=native")
++elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "AppleClang")
++    set(TUNE_FLAG "-mtune=core-avx2")
++    if(CMAKE_OSX_ARCHITECTURES MATCHES "arm64.*")
++        set(TUNE_FLAG "-mtune=apple-m1")
++    endif()
```

Comment:
How do you know it will be the best choice for that specific architecture?

Reply:
Would this match an m2 machine? Then for that for example tuning for `-m1` having `-m2` available would be counterproductive. I think tuning flags might better live in the user's profile, but I can see a usecase for arch specific libs to force it, we just need to be clever on how we handle them

Reply:
@ErniGH  @AbrilRBS 

Sorry. My patch is wrong.
As lemire said, apple-clang >= 15 supports `-march=native`.
https://github.com/lemire/FastPFor/issues/120

I add a condition to `validate()` to drop support apple-clang < 15.0.

Reply:
There is another issue.
In apple-clang 16.0.0, fastpfor/0.2.0 works fine by building by hand(ex. `cmake -B build -S . && cmake --build build`).

There is still compilation errors by building by conan's reipce(ex. `conan create . --build=missing --version=0.2.0`)

Are there any difference?

My .conan2/profile is following.

```
```

Reply:
I'll check tomorrow, thanks for the investigation @toge! :)

Reply:
@toge this might be something on your end, I can compile with your profile just fine, see
```
 $ conan create . --version=0.2.0 -b="&" -s="compiler.cppstd=23"

======== Exporting recipe to the cache ========
fastpfor/0.2.0: Exporting package recipe: /Users/abril/coding/external/toge/recipes/fastpfor/all/conanfile.py
fastpfor/0.2.0: exports: File 'conandata.yml' found. Exporting it...
fastpfor/0.2.0: Calling export_sources()
fastpfor/0.2.0: Copied 1 '.py' file: conanfile.py
fastpfor/0.2.0: Copied 1 '.yml' file: conandata.yml
fastpfor/0.2.0: Copied 1 '.patch' file: 0.2.0-0001-fix-cmake.patch
fastpfor/0.2.0: Exported to cache folder: /Users/abril/.conan2/p/fastp4004cc8b5f8a1/e
fastpfor/0.2.0: Exported: fastpfor/0.2.0#50e29b7b86f8395c0cbc08622c361fbe (2024-11-20 10:07:25 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=23
compiler.libcxx=libc++
compiler.version=16
os=Macos
[platform_tool_requires]
cmake/3.30.3
meson/1.5.2
[replace_tool_requires]
meson/*: meson/[>=1.0 <2]
[conf]


Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=16
os=Macos
[platform_tool_requires]
cmake/3.30.3
meson/1.5.2
[replace_tool_requires]
meson/*: meson/[>=1.0 <2]
[conf]



======== Computing dependency graph ========
Graph root
    cli
Requirements
    fastpfor/0.2.0#50e29b7b86f8395c0cbc08622c361fbe - Cache
    simde/0.8.0#0332093f1457663450843e2da32e99fd - Cache
Build requirements
    meson/1.5.2 - Platform
    pkgconf/2.0.3#f996677e96e61e6552d85e83756c328b - Cache
Replaced requires
    meson/1.2.2: meson/[>=1.0 <2]

======== Computing necessary packages ========
fastpfor/0.2.0: Forced build from source
Requirements
    fastpfor/0.2.0#50e29b7b86f8395c0cbc08622c361fbe:2ef509278305ec0399433950bee509d7cf891fb1 - Build
    simde/0.8.0#0332093f1457663450843e2da32e99fd:da39a3ee5e6b4b0d3255bfef95601890afd80709#a0aab67246db5711c2cfc413808252f8 - Cache
Build requirements
Skipped binaries
    meson/1.5.2, pkgconf/2.0.3

======== Installing packages ========
simde/0.8.0: Already installed! (1 of 2)
fastpfor/0.2.0: Calling source() in /Users/abril/.conan2/p/fastp4004cc8b5f8a1/s/src
fastpfor/0.2.0: Source https://github.com/lemire/FastPFor/archive/refs/tags/v0.2.0.tar.gz retrieved from local download cache
fastpfor/0.2.0: Unzipping v0.2.0.tar.gz to .
fastpfor/0.2.0: Apply patch (conan): enable shared build, disable unittest/utility

-------- Installing package fastpfor/0.2.0 (2 of 2) --------
fastpfor/0.2.0: Building from source
fastpfor/0.2.0: Package fastpfor/0.2.0:2ef509278305ec0399433950bee509d7cf891fb1
fastpfor/0.2.0: Copying sources to build folder
fastpfor/0.2.0: Building your package in /Users/abril/.conan2/p/b/fastp4e6d4fed88d69/b
fastpfor/0.2.0: Calling generate()
fastpfor/0.2.0: Generators folder: /Users/abril/.conan2/p/b/fastp4e6d4fed88d69/b/build/Release/generators
fastpfor/0.2.0: CMakeToolchain generated: conan_toolchain.cmake
fastpfor/0.2.0: CMakeToolchain generated: /Users/abril/.conan2/p/b/fastp4e6d4fed88d69/b/build/Release/generators/CMakePresets.json
fastpfor/0.2.0: CMakeToolchain generated: /Users/abril/.conan2/p/b/fastp4e6d4fed88d69/b/src/CMakeUserPresets.json
fastpfor/0.2.0: CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(simde)
    target_link_libraries(... simde::simde)
fastpfor/0.2.0: Generating aggregated env files
fastpfor/0.2.0: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
fastpfor/0.2.0: Calling build()
fastpfor/0.2.0: Running CMake.configure()
fastpfor/0.2.0: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/Users/abril/.conan2/p/b/fastp4e6d4fed88d69/p" -DSUPPORT_NEON="ON" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/abril/.conan2/p/b/fastp4e6d4fed88d69/b/src"
CMake Deprecation Warning at CMakeLists.txt:6 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- CMAKE_BUILD_TYPE: Release
-- Using Conan toolchain: /Users/abril/.conan2/p/b/fastp4e6d4fed88d69/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: Defining libcxx as C++ flags: -stdlib=libc++
-- Conan toolchain: C++ Standard 23 with extensions OFF
-- The CXX compiler identification is AppleClang 16.0.0.16000026
-- The C compiler identification is AppleClang 16.0.0.16000026
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test SUPPORT_SSE42
-- Performing Test SUPPORT_SSE42 - Failed
-- Performing Test SUPPORT_AVX
-- Performing Test SUPPORT_AVX - Failed
-- Performing Test SUPPORT_AVX2
-- Performing Test SUPPORT_AVX2 - Failed
Building for architecture: arm64
CMake Warning (dev) at CMakeLists.txt:38 (exec_program):
  Policy CMP0153 is not set: The exec_program command should not be called.
  Run "cmake --help-policy CMP0153" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  Use execute_process() instead.
Call Stack (most recent call first):
  CMakeLists.txt:49 (CXX_COMPILER_DUMPVERSION)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- CMAKE_SIZEOF_VOID_P (should be 8): 8
-- Good. You appear to have a 64-bit system. 
-- CMAKE_CXX_COMPILER_ID: AppleClang
-- CMAKE_C_COMPILER: /Library/Developer/CommandLineTools/usr/bin/cc
-- CXX_COMPILER_VERSION: 16.0.0
-- Conan: Target declared 'simde::simde'
-- USING SIMDE FOR SIMD OPERATIONS
-- CMAKE_CXX_FLAGS_DEBUG: -Wall -Wcast-align -ggdb  -std=c++11 -DHAVE_CXX0X -march=native
-- CMAKE_CXX_FLAGS_RELEASE: -Wall -Wcast-align -O3 -DNDEBUG -std=c++11 -DHAVE_CXX0X -march=native
-- CMAKE_C_FLAGS_DEBUG: -Wall -Wcast-align -ggdb  -std=c99 -march=native
-- CMAKE_C_FLAGS_RELEASE: -Wall -Wcast-align -O3 -DNDEBUG -std=c99 -march=native
-- Snappy was found. Building additional targets codecssnappy and inmemorybenchmarksnappy.
-- Configuring done (2.2s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/abril/.conan2/p/b/fastp4e6d4fed88d69/b/build/Release

fastpfor/0.2.0: Running CMake.build()
fastpfor/0.2.0: RUN: cmake --build "/Users/abril/.conan2/p/b/fastp4e6d4fed88d69/b/build/Release" -- -j12
[ 10%] Building CXX object CMakeFiles/FastPFOR.dir/src/bitpacking.cpp.o
[ 20%] Building CXX object CMakeFiles/FastPFOR.dir/src/bitpackingunaligned.cpp.o
[ 30%] Building CXX object CMakeFiles/FastPFOR.dir/src/simdunalignedbitpacking.cpp.o
[ 40%] Building CXX object CMakeFiles/FastPFOR.dir/src/bitpackingaligned.cpp.o
[ 50%] Building C object CMakeFiles/FastPFOR.dir/src/streamvbyte.c.o
[ 60%] Building CXX object CMakeFiles/FastPFOR.dir/src/codecfactory.cpp.o
[ 70%] Building C object CMakeFiles/FastPFOR.dir/src/varintdecode.c.o
[ 80%] Building CXX object CMakeFiles/FastPFOR.dir/src/simdbitpacking.cpp.o
[ 90%] Building CXX object CMakeFiles/FastPFOR.dir/src/horizontalbitpacking.cpp.o

[... a bunch of warnings ...]
      |                                                       ^
40 warnings generated.
39 warnings generated.
[100%] Linking CXX static library libFastPFOR.a
[100%] Built target FastPFOR

fastpfor/0.2.0: Package '2ef509278305ec0399433950bee509d7cf891fb1' built
fastpfor/0.2.0: Build folder /Users/abril/.conan2/p/b/fastp4e6d4fed88d69/b/build/Release
fastpfor/0.2.0: Generating the package
fastpfor/0.2.0: Packaging in folder /Users/abril/.conan2/p/b/fastp4e6d4fed88d69/p
fastpfor/0.2.0: Calling package()
fastpfor/0.2.0: Running CMake.install()
fastpfor/0.2.0: RUN: cmake --install "/Users/abril/.conan2/p/b/fastp4e6d4fed88d69/b/build/Release" --prefix "/Users/abril/.conan2/p/b/fastp4e6d4fed88d69/p"
-- Install configuration: "Release"
[... install output ...]

fastpfor/0.2.0: package(): Packaged 1 file: LICENSE
fastpfor/0.2.0: package(): Packaged 49 '.h' files
fastpfor/0.2.0: package(): Packaged 1 '.a' file: libFastPFOR.a
fastpfor/0.2.0: Created package revision ab241cafba91959a0269bdac3cbec785
fastpfor/0.2.0: Package '2ef509278305ec0399433950bee509d7cf891fb1' created
fastpfor/0.2.0: Full package reference: fastpfor/0.2.0#50e29b7b86f8395c0cbc08622c361fbe:2ef509278305ec0399433950bee509d7cf891fb1#ab241cafba91959a0269bdac3cbec785
fastpfor/0.2.0: Package folder /Users/abril/.conan2/p/b/fastp4e6d4fed88d69/p

======== Launching test_package ========

======== Computing dependency graph ========
Graph root
    fastpfor/0.2.0 (test package): /Users/abril/coding/external/toge/recipes/fastpfor/all/test_package/conanfile.py
Requirements
    fastpfor/0.2.0#50e29b7b86f8395c0cbc08622c361fbe - Cache
    simde/0.8.0#0332093f1457663450843e2da32e99fd - Cache
Build requirements
    meson/1.5.2 - Platform
    pkgconf/2.0.3#f996677e96e61e6552d85e83756c328b - Cache
Replaced requires
    meson/1.2.2: meson/[>=1.0 <2]

======== Computing necessary packages ========
Requirements
    fastpfor/0.2.0#50e29b7b86f8395c0cbc08622c361fbe:2ef509278305ec0399433950bee509d7cf891fb1#ab241cafba91959a0269bdac3cbec785 - Cache
    simde/0.8.0#0332093f1457663450843e2da32e99fd:da39a3ee5e6b4b0d3255bfef95601890afd80709#a0aab67246db5711c2cfc413808252f8 - Cache
Build requirements
Skipped binaries
    meson/1.5.2, pkgconf/2.0.3

======== Installing packages ========
simde/0.8.0: Already installed! (1 of 2)
fastpfor/0.2.0: Already installed! (2 of 2)

======== Testing the package ========
Removing previously existing 'test_package' build folder: /Users/abril/coding/external/toge/recipes/fastpfor/all/test_package/build/apple-clang-16-armv8-23-release
fastpfor/0.2.0 (test package): Test package build: build/apple-clang-16-armv8-23-release
fastpfor/0.2.0 (test package): Test package build folder: /Users/abril/coding/external/toge/recipes/fastpfor/all/test_package/build/apple-clang-16-armv8-23-release
fastpfor/0.2.0 (test package): Writing generators to /Users/abril/coding/external/toge/recipes/fastpfor/all/test_package/build/apple-clang-16-armv8-23-release/generators
fastpfor/0.2.0 (test package): Generator 'CMakeDeps' calling 'generate()'
fastpfor/0.2.0 (test package): CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(FastPFOR)
    target_link_libraries(... FastPFOR::FastPFOR)
fastpfor/0.2.0 (test package): Generator 'CMakeToolchain' calling 'generate()'
fastpfor/0.2.0 (test package): CMakeToolchain generated: conan_toolchain.cmake
fastpfor/0.2.0 (test package): CMakeToolchain generated: /Users/abril/coding/external/toge/recipes/fastpfor/all/test_package/build/apple-clang-16-armv8-23-release/generators/CMakePresets.json
fastpfor/0.2.0 (test package): CMakeToolchain generated: /Users/abril/coding/external/toge/recipes/fastpfor/all/test_package/CMakeUserPresets.json
fastpfor/0.2.0 (test package): Generator 'VirtualRunEnv' calling 'generate()'
fastpfor/0.2.0 (test package): Generating aggregated env files
fastpfor/0.2.0 (test package): Generated aggregated env files: ['conanrun.sh', 'conanbuild.sh']

======== Testing the package: Building ========
fastpfor/0.2.0 (test package): Calling build()
fastpfor/0.2.0 (test package): Running CMake.configure()
fastpfor/0.2.0 (test package): RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/Users/abril/coding/external/toge/recipes/fastpfor/all/test_package" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/abril/coding/external/toge/recipes/fastpfor/all/test_package"
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Using Conan toolchain: /Users/abril/coding/external/toge/recipes/fastpfor/all/test_package/build/apple-clang-16-armv8-23-release/generators/conan_toolchain.cmake
-- Conan toolchain: Defining libcxx as C++ flags: -stdlib=libc++
-- Conan toolchain: C++ Standard 23 with extensions OFF
-- The CXX compiler identification is AppleClang 16.0.0.16000026
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Target declared 'FastPFOR::FastPFOR'
-- Conan: Target declared 'simde::simde'
-- Configuring done (0.4s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/abril/coding/external/toge/recipes/fastpfor/all/test_package/build/apple-clang-16-armv8-23-release

fastpfor/0.2.0 (test package): Running CMake.build()
fastpfor/0.2.0 (test package): RUN: cmake --build "/Users/abril/coding/external/toge/recipes/fastpfor/all/test_package/build/apple-clang-16-armv8-23-release" -- -j12
[ 50%] Building CXX object CMakeFiles/test_package.dir/test_package.cpp.o
[100%] Linking CXX executable test_package
[100%] Built target test_package


======== Testing the package: Executing test ========
fastpfor/0.2.0 (test package): Running test()
fastpfor/0.2.0 (test package): RUN: ./test_package
You are using 0.288 bits per integer. 
```

Reply:
@toge do you want me to merge this, or wait until you can check on your end?

Reply:
@AbrilRBS 
Please go ahead.
This is my local issue.